### PR TITLE
chore: create `TokenCreator` role with permissions to VIEW / CREATE / REVOKE own access tokens

### PR DESCRIPTION
**Pull request by ShreyaLnuHpe**
## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by corban-beaird — APPROVED** [review]
Migration looks solid! 

Do we want to include a release note/doc update in this PR for this new role?

File: docs/manage/security/rbac.rst [hunk 1]
```diff
@@ -532,6 +532,12 @@ assigned globally.
    where ``ROLE_ID`` is the integer role identifier, as listed in ``det rbac list-roles``. To
    disable the assignment of any roles to the newly created workspace, set ``enabled: false``.
 
+``TokenCreator``
+================
+
+The ``TokenCreator`` grants the permissions to create, view, and revoke the user's own access
```
Comment (tara-det-ai) [inline docs/manage/security/rbac.rst:538]:
```suggestion
The ``TokenCreator`` grants users the ability to create, view, and revoke their own access
```

File: docs/release-notes/rbac-new-tokenCreator-role.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
+:orphan:
+
+**New Features**
+
+-  New RBAC role: Added a new ``TokenCreator`` RBAC role. It grants the permissions to create, view,
```
Comment (tara-det-ai) [inline docs/release-notes/rbac-new-tokenCreator-role.rst:5]:
```suggestion
- New RBAC role: Add a ``TokenCreator`` RBAC role, which allows users to create, view,
```

File: docs/release-notes/rbac-new-tokenCreator-role.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+   and revoke the user's own access tokens. It can only be assigned globally.
```
Comment (tara-det-ai) [inline docs/release-notes/rbac-new-tokenCreator-role.rst:6]:
```suggestion
and revoke their own access tokens. This role can only be assigned globally.
```

**Review by tara-det-ai — APPROVED** [review]
LGTM / added suggestions

**Pull request merged**